File: leetcode/2645.构造有效字符串的最少插入数.go
func addMinimum(s string) int {
    ans := int(s[0]) - int(s[len(s) - 1]) + 2
    for i := 1; i < len(s); i ++ {
        ans += (int(s[i]) - int(s[i - 1]) + 2) % 3
    }
    return ans
}

File: leetcode/2645.构造有效字符串的最少插入数.cpp
class Solution {
public:
    int addMinimum(string word) {
        int n = word.size();
        int cnt = 1;
        for(int i = 0; i < n-1; i++){
            if(word[i] >= word[i+1]){
                cnt += 1;
            }
        }
        return 3 * cnt - n;
    }
};